components/checkout/SavedAddressSelectionList.tsx

```diff
@@ -0,0 +1,74 @@
+import clsx from "clsx";
+import React from "react";
+
+import { Spinner } from "@/components";
+import {
+  AddressDetailsFragment,
+  CheckoutError,
+  CountryCode,
+  useUserAddressesQuery,
+} from "@/saleor/api";
+
+import { AddressFormData } from "./AddressForm";
+
+interface SavedAddressSelectionListProps {
+  updateAddressMutation: (address: AddressFormData) => Promise<CheckoutError[]>;
+}
+
+export const SavedAddressSelectionList = ({
+  updateAddressMutation,
+}: SavedAddressSelectionListProps) => {
+  const { loading, error, data } = useUserAddressesQuery();
+  const [selectedSavedAddres, setSelectedSavedAddress] =
+    React.useState<AddressDetailsFragment | null>();
+
+  if (loading) {
+    return <Spinner />;
+  }
+
+  if (error) return <p>Error : {error.message}</p>;
+
+  let addresses = data?.me?.addresses || [];
```

Comment:
```suggestion
  const addresses = data?.me?.addresses || [];
  if (addresses.length === 0) return null;
  return (
    <div className="grid grid-cols-2 mb-2">
    ...
```

I would suggest trying to use the same pattern as the rest of the repo.